infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryServiceAdapter.java

```diff
@@ -0,0 +1,23 @@
+package org.corfudb.infrastructure.logreplication.infrastructure;
+
+import org.corfudb.infrastructure.logreplication.proto.LogReplicationClusterInfo;
+
```

Comment:
I'm not sure if this is really needed, because the logic to prepare for role change or query replication is always the same as it internally faces Corfu... @xiaoqin2012 since you introduced this, can you confirm?

Reply:
In the new PR https://github.com/CorfuDB/CorfuDB/pull/2586, I added the some description for this interface.